### PR TITLE
Expose DFA creation and Lexer's internal DFA to external classes

### PR DESCRIPTION
**Pull request by SharafMohamed**
# Description
- Create additional DFAs that are not stored inside the Lexer
- Allow external classes to get access to the Lexer's internal DFA 

# Validation performed
- Ran reader-parser and buffer-parser with correct output

### CODE REVIEW

File: src/log_surgeon/Lexer.hpp [hunk 1]
```diff
@@ -139,6 +139,19 @@ class Lexer {
...
+            -> std::unique_ptr<finite_automata::RegexDFA<DFAStateType>> const& {
+        return m_dfa;
+    }
+
+    /**
+     * Generate a DFA from an NFA
+     * @param finite_automata::RegexNFA<NFAStateType> nfa
+     * @return std::unique_ptr<finite_automata::RegexDFA<DFAStateType>>
+     */
+    auto nfa_to_dfa(finite_automata::RegexNFA<NFAStateType>& nfa)
```
Comment (kirkrodrigues) [inline src/log_surgeon/Lexer.hpp:152]: We can make this static (along with epsilon_closure), right?

File: src/log_surgeon/Lexer.hpp [hunk 2]
```diff
@@ -139,6 +139,19 @@ class Lexer {
...
+            -> std::unique_ptr<finite_automata::RegexDFA<DFAStateType>> const& {
+        return m_dfa;
+    }
+
+    /**
+     * Generate a DFA from an NFA
+     * @param finite_automata::RegexNFA<NFAStateType> nfa
+     * @return std::unique_ptr<finite_automata::RegexDFA<DFAStateType>>
+     */
+    static auto nfa_to_dfa(finite_automata::RegexNFA<NFAStateType>& nfa)
```
Comment (kirkrodrigues) [inline src/log_surgeon/Lexer.hpp:152]: Our ordering guidelines dictate that static methods come before regular methods (in each section of the class). Can we fix it quickly?

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: src/log_surgeon/Lexer.hpp (hunk 2 above)
Comment (SharafMohamed) [reply src/log_surgeon/Lexer.hpp]: Updated

**Review by kirkrodrigues — APPROVED** [review]
Commit msg: `Expose DFA creation and Lexer's internal DFA. (#16)`?

**Pull request merged**